husarion_ugv_manager/behavior_trees/lights.xml

```diff
@@ -1,85 +1,109 @@
+          <Parallel name="DisplayChargingAnimationSequence"
+                    failure_count="1"
+                    success_count="-1"
+                    _skipIf="battery_health == POWER_SUPPLY_HEALTH_OVERHEAT">
+            <Delay delay_msec="300">
+              <CallSetLedAnimationService name="SetChargerInsertedAnimation"
+                                          id="{CHARGER_INSERTED_ANIM_ID}"
+                                          param=""
+                                          repeating="false"
+                                          service_name="lights/set_animation"
+                                          _skipIf="current_battery_anim_id == CHARGING_BATTERY_ANIM_ID ||&#10;current_battery_anim_id == BATTERY_CHARGED_ANIM_ID"/>
+            </Delay>
+            <Delay delay_msec="2500">
+              <CallSetLedAnimationService name="SetChargingAnimation"
+                                          id="{CHARGING_BATTERY_ANIM_ID}"
+                                          param="{battery_percent_round}"
+                                          repeating="true"
+                                          service_name="lights/set_animation"
+                                          _skipIf="(battery_percent_round == charging_anim_percent  \&#10;&amp;&amp; current_battery_anim_id == CHARGING_BATTERY_ANIM_ID\&#10;|| battery_percent_round == 1.0)"
+                                          _onSuccess="charging_anim_percent = battery_percent_round; \&#10;current_battery_anim_id = CHARGING_BATTERY_ANIM_ID"/>
+            </Delay>
+            <Delay delay_msec="2500">
+              <CallSetLedAnimationService name="SetChargedAnimation"
+                                          id="{BATTERY_CHARGED_ANIM_ID}"
+                                          param="{battery_percent_round}"
+                                          repeating="true"
+                                          service_name="lights/set_animation"
+                                          _skipIf="battery_percent_round != 1.0 \&#10;|| current_battery_anim_id == BATTERY_CHARGED_ANIM_ID"
+                                          _onSuccess="charging_anim_percent = battery_percent_round; \&#10;current_battery_anim_id = BATTERY_CHARGED_ANIM_ID"/>
+            </Delay>
+          </Parallel>
+        </Sequence>
+        <CallSetLedAnimationService name="SerErrorAnimation"
                                     id="{ERROR_ANIM_ID}"
                                     param=""
                                     repeating="true"
                                     service_name="lights/set_animation"
-                                    _skipIf="battery_health != POWER_SUPPLY_HEALTH_OVERHEAT \
-|| current_anim_id == ERROR_ANIM_ID"
+                                    _skipIf="battery_status != POWER_SUPPLY_STATUS_UNKNOWN \&#10;|| current_anim_id == ERROR_ANIM_ID"
                                     _onSuccess="current_anim_id = ERROR_ANIM_ID"/>
```

Comment:
SetErrorAnimation is setting `current_anim_id`, that is also set when manual/ready/estop are triggered. Won't this cause program to keep calling in loop service to set error and manual/ready/estop animations?